editor/src/clj/editor/particlefx.clj

```diff
@@ -437,36 +438,36 @@
 
 (defn- render-emitters-sim [^GL2 gl render-args renderables _rcount]
   (doseq [renderable renderables]
-    (let [{:keys [color emitter-index emitter-sim-data material-attribute-infos max-particle-count vertex-attribute-bytes]} (:user-data renderable)
-          shader (:shader emitter-sim-data)
-          shader-attribute-infos-by-name (shader/attribute-infos shader gl)
-          manufactured-attribute-keys [:position :texcoord0 :page-index :color]
-          shader-bound-attributes (graphics/shader-bound-attributes shader-attribute-infos-by-name material-attribute-infos manufactured-attribute-keys :coordinate-space-world)
-          vertex-description (graphics/make-vertex-description shader-bound-attributes)
-          pfx-sim-request-id (some-> renderable :updatable :node-id)]
-      (when-let [pfx-sim-atom (when (and emitter-sim-data pfx-sim-request-id)
-                                (:pfx-sim (scene-cache/lookup-object ::pfx-sim pfx-sim-request-id nil)))]
-        (let [pfx-sim @pfx-sim-atom
-              raw-vbuf (plib/gen-emitter-vertex-data pfx-sim emitter-index color max-particle-count vertex-description shader-bound-attributes vertex-attribute-bytes)
-              context (:context pfx-sim)
-              vbuf (vtx/wrap-vertex-buffer vertex-description :static raw-vbuf)
-              all-raw-vbufs (assoc (:raw-vbufs pfx-sim) emitter-index raw-vbuf)]
-          ;; TODO: We have to update the "raw-vbufs" here because we will
-          ;;       at some point create the GPU vertex buffer from this call.
-          ;;       This is due to the fact that we don't know how big the vertex buffer
-          ;;       will be until we have a valid GL context: We don't have the correct information
-          ;;       until the shader has been compiled. In the future we could perhaps use the meta-data
-          ;;       we get from the SPIR-V toolchain to figure this out beforehand.
-          (swap! pfx-sim-atom assoc :raw-vbufs all-raw-vbufs)
-          (when-let [render-data (plib/render-emitter pfx-sim emitter-index)]
-            (let [gpu-texture (:gpu-texture emitter-sim-data)
-                  vtx-binding (vtx/use-with context vbuf shader)
-                  blend-mode (convert-blend-mode (:blend-mode render-data))]
-              (gl/with-gl-bindings gl render-args [shader vtx-binding gpu-texture]
-                (shader/set-samplers-by-index shader gl 0 (:texture-units gpu-texture))
-                (gl/set-blend-mode gl blend-mode)
-                (gl/gl-draw-arrays gl GL/GL_TRIANGLES (:v-index render-data) (:v-count render-data))
-                (.glBlendFunc gl GL/GL_SRC_ALPHA GL/GL_ONE_MINUS_SRC_ALPHA)))))))))
+    (let [{:keys [color emitter-index emitter-sim-data material-attribute-infos max-particle-count vertex-attribute-bytes]} (:user-data renderable)]
+      (when-let [shader (:shader emitter-sim-data)]
```

Comment:
Mostly indentation, but I check that the shader exists, so it doesn't spam errors in logs when there is no material.